pyiceberg/catalog/rest.py

```diff
@@ -524,7 +537,32 @@ def create_table(
         except HTTPError as exc:
             self._handle_non_200_response(exc, {409: TableAlreadyExistsError})
 
-        table_response = TableResponse(**response.json())
+        return TableResponse(**response.json())
+
+    @retry(**_RETRY_ARGS)
+    def _create_staged_table(
```

Comment:
The inheritance feels off here. I would just expect to `create_table_transaction`. The current `_create_staged_table` on the `Catalog` is now very specific for non-REST catalogs. Should we introduce another layer in the OOP hierarchy there? Then we can override `create_table_transaction` there.

Reply:
Thanks! This is a great suggestion! After going through the current code, I find that not only `_create_staged_table` but also most helper functions in `Catalog` are specific to non-RestCatalog. Hence, I add a new class named `MetastoreCatalog` (appreciate any suggestions on the name) and make all non-RestCatalog inherit from it instead. 

Since it is a big refactoring, please let me know if you want this to happen in a follow-up PR.